lib/opentelemetry.js

```diff
@@ -0,0 +1,186 @@
+   * span to be auto-detected.
+   * Use this when the current code block does the tracing and it doesn't expect
+   * more tracing to happen down the line.
+   *
+   * @param {String} name span name
+   * @param {api.Span|Boolean} [parentSpan] optional parent span
+   * @param {api.SpanOptions} [options] span options
+   * @returns {api.Span}
+   * @example
+   * // Auto-merge with the currently active span context
+   * const span = self.apos.telemetry.aposStartSpan('event:someEvent');
+   * // Provide a parent span
+   * const span = self.apos.telemetry.aposStartSpan('event:someEvent', parent);
+   * // Do not merge with any parent, start as a root
+   * const span = self.apos.telemetry.aposStartSpan('event:someEvent', false);
+   * // Provide options for the newly created span
+   * const span = self.apos.telemetry.aposStartSpan('event:someEvent', true, {
+   *   attributes: [
+   *     [SemanticAttributes.HTTP_METHOD]: "GET",
+   *     [SemanticAttributes.HTTP_FLAVOR]: "1.1",
+   *     [SemanticAttributes.HTTP_URL]: req.url
+   *   ]
+   * });
+   */
+  function aposStartSpan(name, parentSpan, options) {
+    if ((!parentSpan || parentSpan === true) && parentSpan !== false) {
+      parentSpan = api.trace.getSpan(api.context.active());
+    }
+
+    if (parentSpan) {
+      const ctx = api.trace.setSpan(api.context.active(), parentSpan);
+      return tracer.startSpan(name, options || undefined, ctx);
+    }
+
+    return tracer.startSpan(name, options || undefined);
+  }
+
+  /**
+   * Start span and make it active for all the nested spans.
+   * Use this when the current code block does the tracing, but it also expects
+   * more tracing to happen down the line (async calls).
+   *
+   * @param {String} name span name
+   * @param {Function} fn callback function
+   * @param {api.Span|Boolean} [parentSpan] optional parent span
+   * @param {api.SpanOptions} [options] span options
+   * @returns {api.Span|any} the return value of the callback or the newly created span
+   * @example
+   * // Activate span, return some value
+   * const value = await self.apos.telemetry.aposStartActiveSpan(spanName, async (span) => {
+   *   // Use the span, do work, end span, return any value
+   *   span.end();
+   *   return value;
+   * });
+   * // Activate span, using the context of parent span, return the active span
+   * const span = self.apos.telemetry.aposStartActiveSpan(spanName, async (span) => {
+   *   // Use the span, do work, return the span
+   *   return span;
+   * });
+   */
+  function aposStartActiveSpan(name, fn, parentSpan, options) {
+    if (parentSpan) {
+      const ctx = api.trace.setSpan(api.context.active(), parentSpan);
+      return tracer.startActiveSpan(name, options || undefined, ctx, fn);
+    }
+
+    return tracer.startActiveSpan(name, options || undefined, fn);
+  }
+
+  /**
+   * Invoke a callback within a new context (merged with a parent span).
```

Comment:
Just call it a function rather than a callback (because it doesn't follow the traditional async callback argument pattern and I just don't want to confuse anyone).